src/qibo/ui/mpldrawer.py

```diff
@@ -0,0 +1,687 @@
+                if qbit is tuple:
+                    item += ("q_" + str(qbit[0]),)
+                else:
+                    item += ("q_" + str(qbit),)
+
+            for qbit in gate._control_qubits:
+                if qbit is tuple:
+                    item += ("q_" + str(qbit[0]),)
+                else:
+                    item += ("q_" + str(qbit),)
+
+            gates_plot.append(item)
+
+    return gates_plot
+
+
+def plot(circuit, scale=0.6, cluster_gates=True, style=None):
+    """Main matplotlib plot function for Qibo circuit
+    circuit         A Qibo circuit to plot (type: qibo.models.circuit.Circuit)
+    scale           Scale the ouput plot
+    cluster_gates   Group single-qubit gates
+    style           Style applied to the circuit (built-in styles: garnacha, fardelejo, quantumspain, color-blind and cachirulo)
+
+    ax              An Axes object encapsulates all the elements of an individual plot in a figure (type: matplotlib.axes._axes.Axes)
+    ax.figure       A Figure object (type: matplotlib.figure.Figure)
+    """
+
+    if style is not None:
+        if type(style) is dict:
+            plot_params.update(style)
+        else:
+            plot_params.update(
+                STYLE[style] if style in list(STYLE.keys()) else STYLE["default"]
+            )
+    else:
+        plot_params.update(STYLE["default"])
+
+    inits = list(range(circuit.nqubits))
+
+    labels = []
+    for i in range(circuit.nqubits):
+        labels.append("q_" + str(i))
+
+    if len(circuit.queue) > 0:
+
+        all_gates = []
+        for gate in circuit.queue:
+            if isinstance(gate, gates.FusedGate):
+                min_q, max_q = _get_min_max_qbits(gate)
+
+                fgates = None
+
+                if cluster_gates:
+                    fgates = _make_cluster_gates(_process_gates(gate.gates))
+                else:
+                    fgates = _process_gates(gate.gates)
+
+                l_gates = len(gate.gates)
+                equal_qbits = False
+                if min_q != max_q:
+                    l_gates = len(fgates)
+                else:
+                    max_q += 1
+                    equal_qbits = True
+
+                all_gates.append(
+                    FusedStartGateBarrier(min_q, max_q, l_gates, equal_qbits)
+                )
+                all_gates += gate.gates
+                all_gates.append(FusedEndGateBarrier(min_q, max_q))
+            else:
+                all_gates.append(gate)
+
+        gates_plot = _process_gates(all_gates)
+
+        if cluster_gates:
+            gates_cluster = _make_cluster_gates(gates_plot)
+            ax = _plot_quantum_schedule(gates_cluster, inits, labels, scale=scale)
+            return ax, ax.figure
+
+        ax = _plot_quantum_circuit(gates_plot, inits, labels, scale=scale)
+        return ax, ax.figure
+    else:
+        ax = _plot_lines_circuit(inits, labels, scale=scale)
+        return ax, ax.figure
```

Comment:
It would be ideal if a circuit with no gates would not require a dedicated function. I.e.:
- for sure `all_gates` would correctly be `[]` after the `for` loop (which would not enter)
- `_process_gates([])` will also return `[]`
- `_plot_quantum_circuit([], ...)` should be able to do the final job :)

Reply:
Maybe, it might be sufficient to promote some `ng` to `nq+ng` in `_plot_quantum_circuit()`

Reply:
Following your suggestion, I also review the code in `_plot_quantum_schedule` to be just a single call to `_plot_quantum_circuit` this way I can get rid of `_plot_lines_circuit` and make `_plot_quantum_circuit` the main internal function to draw the circuits